\chapter*{Abstract (English)}

Single cell analysis provide a new way to inspect biological process at single cell resolution. Recently, new sequencing protocols are developed to simultaneously profile transcriptome, epigenome, and proteome features in single cell. With these technology, researchers are able to interprete the biological like gene regulation, transcript factor binding events in a more effective way. The analysis of these simultaneous profiling data, namely, single cell mulitmodal analysis get more and more popular.

However, a major challenge of single cell multimodal analysis is that the feature space of distinct mulitmodalties are extremely different. The different feature sizes, the sparsity and the distributions pose a huge challenge to utilize the features to obtain a shared uniform latent space across all modalities. Although several methods have been proposed to address the issue, these methods either need too much prior knowledge to set the parameters or lack of scalebility. Moreover, many methods only work for specific modality types or it's impossible to interprete the inferred latent components.


Another critical issue when dealing with single cell mulitmodal is to infer trajectories to capture the cell lineage development. There are many methods have been developed to infer trajectories for single cell data. However, rarely methods are designed for single cell multimodal analysis which can offer more information for the inference. Moreover, many trajectory analysis methods only utlize lower dimensions to infer trajectories leaving high dimensional information unused. Furthermore, when encounter complex datasets with too many lineages, many methods are not capable to infer appriproiate trajectories. What's more, most of methods only use the single of nodes, lack of using the signals on edges.


In this thesis, we propose MOJITOO and PHLOWER account respectively for single cell mulitmodal integration and that of trajectory inference. MOJITOO is developed to address the issue aferomentioned using Cannonical Correlation Analysis(CCA). MOJITOO is able to integate any types of modality combinations effectively and efficiently. Moreover, MOJITOO requires no prior knowledge to setup the parameters by only capture the most share latent space. We performed comprehansive evaluation benchmarking to check the space preservation, clustering and distance accuracy to find MOJITOO perform the best overall. Moreover, demonstrate the latent space MOJITOO obtained can capture major cell types or cell type differentiation information.

And PHLOWER is designed to address the trajectory inference issues that current methods lack of. It's designed for trajectory inference for single cell multimodal data using hodge laplacian decomposition. We show that PHLOWER score all the first place in four metrics for the comprehansive trajectory inferece evaluation framework using data with different complexities. Moreover we also explore the power of mulitmodal analysis using PHLOWER on large-scale single cell mulitmodal with transcriptome and epigenome from single cell kidney organoid data. We are able to detect transcription factors that regulated a lineage gene expression. Our analyses shed novel light on mechanisms of kidney lineage development. Altogether, these results demonstrate that PHLOWER is a powerful computational approach in biological studies for single cell mulitmodal analysis.

