\chapter*{Publications}
As required by \S 5(3) of the \\

\begin{addmargin}{0.05\textwidth}
\textit{Promotionsordnung f\"ur die Fakult\"at f\"ur Mathematik, Informatik und Naturwissenschaften der Rheinisch-Westf\"alischen Technischen Hochschule Aachen vom 27.09.2010 in der \linebreak Fassung der zweiten Ordnung zur \"anderung der Promotionsordnung vom 30.06.2014 \linebreak (ver\"offentlicht als Gesamtfassung)},
\end{addmargin} 
\vspace{0.5cm}

\noindent
a declaration of results that are published by the author as well as particular contributions to co-authored publications follows. \\


I am the main author of the following publication, which includes all the results presented in this
thesis:

\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item \textbf{M. Cheng}, Z. Li, and I.G. Costa. MOJITOO: a fast and universal method for integration of multimodal single-cell data. \textbf{Bioinformatics} (2022) \\
	\end{itemize}
\end{addmargin}


\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item \textbf{M. Cheng}, J. Jansen, K. Reimer, J. Nagai, Z. Li, M. Grasshoff, M. T. Schaub, C. Kuppe, R. Kramann,  I. G. Costa. PHLOWER - Single cell trajectory analysis using Decomposition of the Hodge Laplacian. \textbf{under revision} (2024)\\

	\end{itemize}
\end{addmargin}


I am also the co-first author of the following publication:\\


\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item N. Kaesler, \textbf{M. Cheng}, J. Nagai, J. O’Sullivan, F. Peisker, E.M. Bindels, A. Babler, J. Moellmann, P. Droste1, G. Franciosa, A. Dugourd, J. Saez-Rodriguez, S. Neuss, M. Lehrke, P. Boor, C. Goettsch, J.V. Olsen, T. Speer, T. Lu, K. Lim,
J. Floege, L. Denby, I.G. Costa, R. Kramann Mapping cardiac remodeling in chronic kidney disease. \textbf{Sci. Adv.} (2023)\\

	Nadine, et al. utilized single cell gene expression profiling in two distinct mouse models, namely subtotal nephrectomy~(STNx) and ischemia-reperfusion injury~(IRI), to dissect the intricate molecular and cellular mechanisms underlying cardiac remodeling in chronic kidney disease (CKD). \\
	\end{itemize}
\end{addmargin}

\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item M. Buse, \textbf{M. Cheng}, V. Jankowski, M. Lellig, V. Sterzer, T. Strieder, K. Leuchtle, I. V. Martin, C. Seikrit, P. Brinkkoettter, G. Crispatzu, J. Floege, P. Boor, T. Speer, R. Kramann, T. Ostendorf, M. J. Moeller, I. G. Costa, E.Stamellou. Lineage tracing reveals transient phenotypic adaptation of tubular cells during acute kidney injury. \textbf{iScience} (2024)  \\

	Buse, et al. employed single-cell gene expression profiling at multiple time points throughout the progression of kidney disease to unravel the fate of differentiating proximal tubular cells (PT) and their transition into distinct injured. \\
	\end{itemize}
\end{addmargin}



\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item D. Schumacher, \textbf{M. Cheng}, C. Kuppe, F. Peisker, S. Ziegler, E.M. Bindels, R.K. Schneider,  Z. Li,  I. G. Costa, R. Kramann. Epigenetic landscape of myocardial infarction points towards a role of pericytes in cardiac remodeling. \textbf{Nature Cardiovascular Research, under revision} (2024) \\

	David, et al. utilized single-cell gene expression, chromatin accessibility and spatial transcriptome to map gene regulatory changes  at different time points after myocardial infarction in mice to understanding of cellular and transcriptional changes after myocardial infarction.\\
	\end{itemize}
\end{addmargin}



Finally, the following publications did not directly contribute to this thesis but shaped a general understanding of next-generation sequencing analysis and gene regulation of scRNA and scATAC analysis and machine learning. They were published during my Ph.D. studies at RWTH University Aachen.

\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item Z. Li, C. Kuppe, S. Ziegler, \textbf{M. Cheng}, N. Kabgani, S. Menzel, M. Zenke, R. Kramann, I.G. Costa. Chromatin-accessibility estimation of single-cell ATAC data with scOpen. \textbf{Nat. Commun.} (2021)
	\end{itemize}
\end{addmargin}

\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item C. Kuppe, R. O R. Flores, Z. Li, M. Hannani, J. Tanevski, M. Halder, \textbf{M. Cheng}, S. Ziegler, X. Zhang, F. Preisker, N. Kaesler, Y. Xu, R. M. Hoogenboezem, E. M. Bindels, R. K. Schneider, H. Milting, I.G. Costa, J. Saez-Rodriguez, R. Kramann. Spatial multi-omic map of human myocardial infarction. \textbf{Nature} (2021).
    \end{itemize}
\end{addmargin}

\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item M. Vucur,  A. Ghallab, A.T. Schneider, A. Adili,  \textbf{M. Cheng},  M.Castoldi, M.T. Singer,  V. Büttner, L.S. Keysberg, L. Küsgens, I. G. Costa, M. Kohlhepp.  Sublethal necroptosis signaling promotes inflammation and liver cancer. \textbf{Immunity} (2023).
	\end{itemize}
\end{addmargin}

\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item  D.L. Hölscher, N. Bouteldja, M. Joodaki, M.L. Russo, Y.C. Lan, A.V. Sadr, \textbf{M. Cheng}, V. Tesar, S.V. Stillfried, B.M. Klinkhammer, J. Barratt. Next-Generation Morphometry for pathomics-data mining in histopathology. \textbf{Nat. Commun.} (2023).
	\end{itemize}
\end{addmargin}

\begin{addmargin}{0.02\textwidth}
	\begin{itemize}
	\item M. Joodaki, M. Shaigan, V, Parra, R.D. Bülow, C. Kuppe, D.L. Hölscher, \textbf{M. Cheng}, J. Nagai, M. Goedertier, N. Bouteldja, V. Tesar, J. Barratt, I. Roberts, R. Coppo, R. Kramann, P. Boor, I.G. Costa. Detection of PatIent-Level distances from single cell genomics and pathomics data with Optimal Transport (PILOT). \textbf{Mol. Syst. Biol.} (2023)
	\end{itemize}
\end{addmargin}


I implemented MOJITOO \& PHLOWER and built the benchmarking dataset. All figures and tables in this thesis are authored by me and exceptions were explicitly stated in their respective captions. I performed all computational analyses in this study with the aid of Prof. Dr. Ivan G. Costa. All chapters of this thesis have been written by me. Prof. Dr. Ivan G. Costa provided support in all stages of this research including thesis manuscript writing. Mostly out of habit and to honor the fact that research is rarely an entirely solitary process I will, in this thesis, rely on the use of the first-person plural pronoun ``we'' in the text, as a nosism.
