%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Chapter 1 -> Introduction
% Author: Mingbo Cheng
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Introduction}
\label{chapter:introduction}

\graphicspath{{chapter1/figs/}}

\section{Motivation}
\label{introduction:sec1.motivation}

Simultaneously studying multimodalities at a single cell like joint profiling chromatin accessibility and gene expression at a single-cell level offers the potential to unravel networks of gene regulation powered by enhancers. Biological processes and systems exhibit dynamism. To comprehensively grasp the molecular and cellular constituents, along with the networks they form during these processes, researchers frequently gather data across temporal intervals. However, there are many obstacles in the process of analyzing single cell multi-modal data. The first challenge is the integration of these modalities is challenging due to the different sparsity of the data and the different characteristics of these modalities. Existing methods for the integration of single-cell multi-omics data are mostly based on matrix factorization, which is not suitable for the integration of sparse data. In addition, these methods do not consider the different characteristics of the two modalities, which may lead to the loss of information. Therefore, there is a need for a new method to integrate single-cell multi-omics data.

Cells undergo dynamic molecular state changes in various situations, such as during development, the cell cycle, the circadian rhythm, and regeneration. However, the destructive nature of these assays poses challenges in studying continuous processes, where measuring the same cell multiple times is ideal. To address this challenge, numerous mathematical methods have emerged, utilizing ensembles of single cells at different internal states to reconstruct cell fate development, a concept known as trajectory inference. More than 70 methods have been developed to infer trajectories that indicate the complexity of the area. 


\begin{figure}[!ht]
	\centering
	\includegraphics[width=0.95\textwidth]{multimodal_schematic/fig}
	\vspace{0.1cm}
	\caption[single cell multi-modal from sequencing to integration \& trajectory inference]{single cell multi-modal from sequencing to integration \& trajectory inference}
	\label{fig:sc_mulitmodal_sequencing_to_integration_ti}
\end{figure}



Thesis Overview
\section{Contributions}
\label{introduction:sec2.contributions}

\begin{itemize}
	\item \textbf{A Novel computational method for single cell multi-modal integration:} We derived a computational approach based on canonical correlation analysis~(CCA) and correlation test to integrate single cell multi-modal data. The experiment have shown that our method can effectively integrate multiple modalities of single cell to recover the most common signal embedded accoss modalities.

	\item \textbf{A comprehensive evaluation of multi-modal integration methods:} We proceeded a comprehensive evaluation for single-cell multi-modal integration, involving: (1) our novel method; (2) seven state-of-the-art integration methods for single cell multi-modalities; In addition, we collected six single cell multi-modal datasets where true labels are available. Moreover, we evaluated the integration results using three metrics as well as the computational resources consuming. Our evaluation represents the most comprehensive comparison for single cell multi-modal integration.

	\item \textbf{A Novel computational method for single cell trajectory inference:} We also derived a computational method based on Delaunay triangulation and graph hodge laplacian decomposition to infer a cell fate tree by performing clustering on paths of single cell data. The experiment have revealed that our approach can accurately detect different branches of single cell data especially for those has complex structured tree single cell data.

	\item \textbf{A series of multiple branches datasets for trajectory inference:} Inspired by PHATE~\citep{moon2017phate}, we constructed 10 datasets with branches number range from 3 to 12 using DLA tree algorthm for the evaluation of different levels of branches single cell data. We further plugged our data into dynverse trajectory inference evaluation system, the datasets covered diverse complexity of single cell data, can be used to further evaluated any new trajectory inference approaches.

	\item \textbf{Evaluation of single cell trajectory inference:} We implmented our method for dynverse evaluation, and also another method STREAM~\citep{chen2019stream} which hasn't been invovled by dynverse. We comprehensively compare our methods with all compared methods.

	\item \textbf{Biological validation with novel single cell Multi-modal data:} We apply our methods MOJITOO and PHLOWER respectively to integrate and infer trajectories for noval single cell multi-modal kidnay organoid data. Our method can effectively integrate the data and also created cell fate tree.
\end{itemize}

\section{Thesis Structure}
\label{introduction:sec3.structure}
In Chapter 2, we introduce all the concepts necessary for understanding this thesis. We describe single cell profiling technology, which measures the transcriptome, epigenome, and enables simultaneous profiling of up to three types of features. Additionally, we present computational workflows for single cell RNA and single cell ATAC-seq analysis. Finally, we point out the computational challenges in the analysis of single cell multimodal data, especially in the integration of single cell multimodal data and trajectory inference, reviewing published approaches comprehensively~(\fref{fig:thesis_overview}B).
\begin{figure}[!ht]
	\centering
	\includegraphics[width=0.95\textwidth]{thesis_overview/fig}
	\vspace{0.1cm}
	\caption[thesis overview]{thesis overview}
	\label{fig:thesis_overview}
\end{figure}
In Chapter 3, we present the proposed computational approach for performing the integration of single cell multimodal data. Our method is based on Canonical Correlation Analysis~(CCA)~(\fref{fig:thesis_overview}C).

In Chapter 4, we describe the experiments and results conducted for single cell multimodal integration to evaluate our method. We first introduce the benchmarking data, which is composed of several real-world scATAC-seq datasets. Next, we describe the computational strategies to benchmark the performance of our approach from different perspectives. Moreover, we outline all analysis results based on our experiments~(\fref{fig:thesis_overview}D).

In Chapter 5, we present the proposed computational approach for trajectory inference. Our method is based on Normalized Hodge Laplacian Decomposition. We use the signal on edges to group trajectory paths and eventually generate a cell differentiation tree~(\fref{fig:thesis_overview}E).

In Chapter 6, we describe the experiments and results conducted for single cell trajectory inference to evaluate our method. We first describe the simulation data construction methods. Next, we introduce a trajectory inference framework for the evaluation, which can benchmark the performance of our approach from different perspectives. Then, we present the results of the benchmarking on the simulated dataset. Moreover, we use a real-world single cell multimodal dataset for biological analysis, outlining the experiment design and the insights obtained from our trajectory inference methods. Finally, we describe the validation results in the biological aspect~(\fref{fig:thesis_overview}F).

Finally, we conclude this thesis in Chapter 7 by discussing the presented results, highlighting key findings, and identifying potential avenues for future research~(\fref{fig:thesis_overview}G).
